apps/server/lib/lexical/server/provider/handlers/code_lens.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Lexical.Server.Provider.Handlers.CodeLens do
+  alias Lexical.Document
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.Project
+  alias Lexical.Protocol.Requests
+  alias Lexical.Protocol.Responses
+  alias Lexical.Protocol.Types.CodeLens
+  alias Lexical.Server.Provider.Env
+  alias Lexical.Server.Provider.Handlers
+
+  import Document.Line
+  require Logger
+
+  def handle(%Requests.CodeLens{} = request, %Env{} = env) do
+    lenses =
+      case reindex_lens(env.project, request.document) do
+        nil -> []
+        lens -> List.wrap(lens)
+      end
+
+    response = Responses.CodeLens.new(request.id, lenses)
+    {:reply, response}
+  end
+
+  defp reindex_lens(%Project{} = project, %Document{} = document) do
+    if Path.basename(document.path) == "mix.exs" do
```

Comment:
Should this also check that indexing is enabled?

Reply:
Good point